pydomcfg/domzgr/zgr.py

```diff
@@ -0,0 +1,154 @@
+"""
+Base class to generate NEMO v4.0 vertical grids.
+"""
+
+from itertools import product
+
+import numpy as np
+from xarray import DataArray, Dataset
+
+
+class Zgr:
+    """
+    Base class to generate NEMO vertical grids.
+    """
+
+    def __init__(self, ds_bathy: Dataset, jpk: int):
+        """
+        Initialize class.
+
+        Parameters
+        ----------
+        ds_bathy: Dataset
+             xarray dataset including grid coordinates and bathymetry
+        jpk: int
+             number of computational levels
+        """
+
+        self._bathy = ds_bathy
+        self._jpk = jpk
+
+    # -------------------------------------------------------------------------
+    def init_ds(self) -> Dataset:
+        ds = self._bathy.copy()
+        jpi = ds.dims["x"]
+        jpj = ds.dims["y"]
```

Comment:
I'd use sizes rather than dims. It does the same, but I think it's better because the behavior of `.dims` with DataArrays is not consistent.
```suggestion
        jpi = ds.sizes["x"]
        jpj = ds.sizes["y"]
```

Reply:
Got it